README.markdown

```diff
@@ -90,6 +90,13 @@ This generates:
 
 Tasks can be added only if they do not depend on `sourceGenerators`. Otherwise, it will cause an infinite loop.
 
+**CAVEAT**: Adding tasks in general to `buildInfoKeys` can have bad side-effects due to how sbt-buildinfo uses them. See [#87][] / [scala-js/scala-js#2363][] / [sbt/sbt#2970][] / [sbt/sbt#2156][].
```

Comment:
I'd phrase it as something like:

The tasks added to `buildInfoKeys` are not deduplicated against sbt's task engine. Therefore, calling complex tasks that requires locking and/or exclusive access may cause unexpected race condition. Keep it to relatively simple tasks.